docs/source/java/dataset.rst

```diff
@@ -228,3 +250,25 @@ native objects after using. For example:
     AutoCloseables.close(factory, dataset, scanner);
 
 If user forgets to close them then native object leakage might be caused.
+
+Development Guidelines
+======================
+
+* Related to the note about ScanOptions batchSize argument: Let's try to read a Parquet file with gzip compression and 3 row groups:
+
+    .. code-block::
+
+       # Let configure ScanOptions as:
+       ScanOptions options = new ScanOptions(/*batchSize*/ 32768);
+
+       $ parquet-tools meta data4_3rg_gzip.parquet
+       file schema: schema
+       age:         OPTIONAL INT64 R:0 D:1
+       name:        OPTIONAL BINARY L:STRING R:0 D:1
+       row group 1: RC:4 TS:182 OFFSET:4
+       row group 2: RC:4 TS:190 OFFSET:420
+       row group 3: RC:3 TS:179 OFFSET:838
+
+    In this case, we are configuring ScanOptions batchSize argument equals to
```

Comment:
Please could you help me to understand your point. It is how it is working (i.e. every row group has 4 rows or less, this value are compared with the batchSize internally). You are proposing to change the example? change the paragraph?

Reply:
32768 isn't a request for rows, it's a limit. Since the file only has 4 rows in in the first row group, the actual batch size will be 4. If the file had more than 32768 rows, then it would get split into blocks of 32768 rows or less.

Reply:
Thank you for that detail, let me add that detail